src/mds/FSMap.cc

```diff
@@ -224,6 +224,51 @@ void FSMap::print(ostream& out) const
   }
 }
 
+void FSMap::print_daemon_summary(ostream& out) const
+{
+  // this appears in the "services:" section of "ceph status"
+  int num_up = 0, num_in = 0, num_failed = 0;
+  for (auto& [fscid, fs] : filesystems) {
+    num_up += fs->mds_map.get_num_up_mds();
+    num_in += fs->mds_map.get_num_in_mds();
+    num_failed += fs->mds_map.get_num_failed_mds();
+  }
+  int num_standby = standby_daemons.size();
+  out << num_up << "/" << num_in << " daemons up";
+  if (num_failed) {
+    out << " (" << num_failed << " failed)";
+  }
+  if (num_standby) {
+    out << ", " << num_standby << " standby";
+  }
+}
+
+void FSMap::print_fs_summary(ostream& out) const
+{
+  // this appears in the "data:" section of "ceph status"
+  if (!filesystems.empty()) {
+    int num_degraded = 0, num_stopped = 0, num_healthy = 0;
+    for (auto& [fscid, fs] : filesystems) {
+      if (fs->mds_map.is_degraded()) {
```

Comment:
I'd suggest maybe breaking out into damaged (requires admin rescue) and degraded (?? can we think of an alternate word) service (missing ranks).

Another one is `down` when `max_mds==0`.

Reply:
``max_mds==0`` means the fs is deliberately offline, right?  this is what I thought 'stopped' was... adjusted the condition.

I don't see a way to distinguish between damaged and recovering... :/

Reply:
> max_mds==0 means the fs is deliberately offline, right?

Yes. `fs set <foo> down true`

> I don't see a way to distinguish between damaged and recovering... :/

I think we need to add a `MDSMap::get_damaged_mds_set` to see damaged ranks. `get_failed_mds_set` will tell you if the fs is "failed".

Reply:
I went with healthy, recovering, failed, and stopped.  And then damaged is separate.  Sound ok?

Reply:
Sounds good.